koin-projects/koin-core/src/main/kotlin/org/koin/core/scope/Scope.kt

```diff
@@ -260,17 +260,19 @@ data class Scope(
      * (single definition of th current scope is root)
      *
      * @param instance The instance you're declaring.
+     * @param clazz The primary type for this declaration
      * @param qualifier Qualifier for this declaration
      * @param secondaryTypes List of secondary bound types
      * @param override Allows to override a previous declaration of the same type (default to false).
      */
     fun <T : Any> declare(
```

Comment:
Instead of having to provide a new class, we can directly rely on reified type: `inline fun <reified T : Any>`